### PR TITLE
fix metalink 'size' field parsing

### PR DESCRIPTION
**Pull request by oliverkurth**
Fix metalink `size` field parsing. Alternative to PR #489 .

### CODE REVIEW

File: plugins/metalink/utils.c [hunk 1]
```diff
@@ -467,28 +466,30 @@ TDNFXmlParseData(
...
     if(!strcmp(elementInfo->startElement, TAG_NAME_FILE))
     {
         elementInfo->dwError = TDNFParseFileTag(userData);
         BAIL_ON_TDNF_ERROR(elementInfo->dwError);
     }
     else if(!strcmp(elementInfo->startElement, TAG_NAME_SIZE))
     {
-        //Get File Size.
-        TDNFAllocateStringN(val, len, &size);
+        char size_buf[12], *p = size_buf;
```
Comment (tapakund) [inline plugins/metalink/utils.c:480]: could you explain why did we hardcode the size_buf to 12 bytes?

Comment (oliverkurth) [reply plugins/metalink/utils.c:480]: It's more than enough to hold a 32 bit integer, which in turn is way more than needed for the file size of `repomd.xml`.

File: plugins/metalink/utils.c [hunk 2]
```diff
@@ -467,28 +466,30 @@ TDNFXmlParseData(
...
+        const char *q = val;
 
-        if(!size)
+        while (*q && q < val + len && p < size_buf + sizeof(size_buf) && isdigit(*q))
```
Comment (oliverkurth) [inline plugins/metalink/utils.c]: This should actually be `p < size_buf + sizeof(size_buf) - 1`. I'll fix it later.

Comment (oliverkurth) [reply plugins/metalink/utils.c]: Done.

**Pull request merged**